### PR TITLE
Add icon for flow implementation. Display proper icon based on selected

### LINKED ISSUE #2440: No icons for flows in outline view
## Summary

The outline view shows default EMF generated icons for flow specifications, flow implementations, and end to end flows. We have proper icons in `org.osate.aadl2.edit`, which should be used. An icon for end to end flow needs to be created.

## Expected and Current Behavior

image

## Environment

* **OSATE Version**: 2.8.0
* **Operating System**: all

### PR DESCRIPTION
**Pull request by asazonova**
Add icon for flow implementation. Display proper icon based on selected specification implementation: display flow path, sink or source

Fixes #2440

### CODE REVIEW

**Comment by asazonova** [comment]
I also changed FlowSink.gif  file only because it shows up with white background in dark theme view (all others have transparent background, so they show up with black background). New file also has transparent background to match the others

File: core/org.osate.aadl2.edit/src/org/osate/aadl2/provider/FlowImplementationItemProvider.java [hunk 1]
```diff
@@ -136,9 +140,19 @@ protected EStructuralFeature getChildFeature(Object object, Object child) {
...
 	 */
+	@Override
 	public Object getImage(Object object) {
+		if (((FlowImplementationImpl) object).getInEnd() == null
+				&& ((FlowImplementationImpl) object).getOutEnd() == null) {
+			return null;
+		} else if (((FlowImplementationImpl) object).getInEnd() == null) {
+			return overlayImage(object, getResourceLocator().getImage("full/obj16/FlowSource"));
+		} else if (((FlowImplementationImpl) object).getOutEnd() == null) {
+			return overlayImage(object, getResourceLocator().getImage("full/obj16/FlowSink"));
```
Comment (lwrage) [inline core/org.osate.aadl2.edit/src/org/osate/aadl2/provider/FlowImplementationItemProvider.java:153]: Add icons for flow source/sink implementation (and I above the line like flow implementation).

File: core/org.osate.aadl2.edit/src/org/osate/aadl2/provider/FlowImplementationItemProvider.java [hunk 2]
```diff
@@ -136,9 +140,19 @@ protected EStructuralFeature getChildFeature(Object object, Object child) {
...
+		}
+
 		return overlayImage(object, getResourceLocator().getImage("full/obj16/FlowImplementation"));
```
Comment (lwrage) [inline core/org.osate.aadl2.edit/src/org/osate/aadl2/provider/FlowImplementationItemProvider.java:156]: For consistency name this file FlowPathImplementation.gif

**Review by lwrage — CHANGES_REQUESTED** [review]
Please add an icon for end to end flow.

**Review by lwrage — APPROVED** [review]

**Comment by lwrage** [comment]
This PR doesn't fix #2442 because the labels are still either _Flow Specification_ or _Flow Implementation_. I'll merge it to close #2440 and move #2442 back to "Next".

**Pull request merged**